### PR TITLE
[traderepublic] Added another format for dividends

### LINKED ISSUE #1105: [ TradeRepublic ] Import special dividend (SL Green Realty)
der Import der Sonderdividende von SL Green Realty schlägt bei TradeRepublic fehl.

Grüße 
Christian

## Document
<details><summary>21-01-D-SLgreen_Sonder-Abrechnung.pdf</summary><p>

```json
[
  [
    "TRADE REPUBLIC BANK GMBH",
    "KASTANIENALLEE 32",
    "10435 BERLIN",
    "Atcyynrqb Aheegsxsgn",
    "Lqyzzyqjofidr 48",
    "09052 Glzcyag",
    "SEITE",
    "1",
    "von",
    "1",
    "DATUM",
    "20.01.2021",
    "DEPOT",
    "5665216682",
    "Trade Republic Bank GmbH",
    "Kastanienallee 32",
    "10435 Berlin",
    "www.traderepublic.com",
    "lyfoeqa@eribtwtrngizx.xxc",
    "Sitz der Gesellschaft: Düsseldorf",
    "AG Düsseldorf HRB 85864",
    "USt-ID DE307510626",
    "Geschäftsführer",
    "Andreas Willius",
    "Karsten Müller",
    "ABRE",
    "/ 21.01.2021",
    "/ 72070701",
    "/ 960d-2377",
    "REINVESTIERUNG",
    "ÜBERSICHT",
    "NR.",
    "BUCHUNG",
    "WERTPAPIER",
    "BETRAG",
    "1",
    "Reinvestierung",
    "SL Green Realty Corp.",
    "Registered Shares DL -,01",
    "US78440X1019",
    "9 Stk.",
    "2",
    "Reinvestierung",
    "SL Green Realty Corp.",
    "Registered Shares DL -,01",
    "US78440X1019",
    "Stk.",
    "ABRECHNUNG",
    "NR.",
    "POSITION",
    "BETRAG",
    "1",
    "Bruttoertrag",
    "14,64 USD",
    "Quellensteuer für US-Emittent",
    "-2,20 USD",
    "GESAMT",
    "12,44 USD",
    "2",
    "Barausgleich",
    "14,64 USD",
    "Quellensteuer gem. Ertrag",
    "-2,20 USD",
    "Zwischensumme",
    "12,44 USD",
    "Zwischensumme",
    "1,20962 EUR/USD",
    "10,28 EUR",
    "GESAMT",
    "10,28 EUR",
    "BUCHUNG",
    "VERRECHNUNGSKONTO",
    "VALUTA",
    "BETRAG",
    "JL24701691426964796144",
    "15.01.2021",
    "22,72 USD",
    "SL Green Realty Corp. Registered Shares DL -,01 in Wertpapierrechnung in Deutschland.",
    "Lagerland: Vereinigte Staaten",
    "Diese Abrechnung wird maschinell erstellt und daher nicht unterschrieben.",
    "Sofern keine Umsatzsteuer ausgewiesen ist, handelt es sich gem. § 4 Nr. 8 UStG um eine umsatzsteuerfreie",
    "Leistung."
  ]
]

```

</p></details>

## Community profile

https://community.tresor.one/u/ca6e3192

---
Issue submitted with https://import.tresor.one

### LINKED ISSUE #1134: Trade Republic es können keine Aktivitäten ausgelesen werden
## Document
<details><summary>Kontoauszug.pdf</summary><p>

```json
[
  [
    "TRADE REPUBLIC BANK GMBH",
    "KASTANIENALLEE 32",
    "10435 BERLIN",
    "Oksujv Dblhzkaxf",
    "Böihtpkp. 92J",
    "R 82975 Nrzfiyak O. CRWN. YU Dinqp",
    "SEITE",
    "1",
    "von",
    "1",
    "DATUM",
    "09.10.2020",
    "DEPOT",
    "916448283",
    "Trade Republic Bank GmbH",
    "Kastanienallee 32",
    "10435 Berlin",
    "www.traderepublic.com",
    "azfxace@uygwiuxdhqevk.bms",
    "Sitz der Gesellschaft: Düsseldorf",
    "AG Düsseldorf HRB 85864",
    "USt-ID DE307510626",
    "Geschäftsführer",
    "Andreas Willius",
    "Karsten Müller",
    "KONTOAUSZUG",
    "zum 30.09.2020",
    "BUCHUNGEN",
    "Prfnrjryblp üwcm hkq Phkqpwnpc pj Cdkdgu Afpgkggpjbjmqwabl XJ59618922919422053307.",
    "BUCHUNGSTAG /",
    "WERTSTELLUNG",
    "BUCHUNGSTEXT",
    "BETRAG IN EUR",
    "02.07.2020",
    "06.07.2020",
    "Ausführung Handel Direktkauf Kauf BE6697996746 ACGHUJCL QA",
    "6871090220200702",
    "-12,55",
    "06.07.2020",
    "08.07.2020",
    "Ausführung Handel Direktverkauf Verkauf US6541101050 NIKOLA CORP.",
    "8331554420200706",
    "51,20",
    "07.07.2020",
    "07.07.2020",
    "Auszahlung an Referenzkonto",
    "-52,28",
    "07.07.2020",
    "09.07.2020",
    "Ausführung Handel Direktverkauf Verkauf QV6572415751 OCLQDLZV OX",
    "5669521120200707",
    "5,24",
    "07.07.2020",
    "07.07.2020",
    "Auszahlung an Referenzkonto",
    "-5,24",
    "25.09.2020",
    "25.09.2020",
    "Referral bonus",
    "15,00",
    "30.09.2020",
    "30.09.2020",
    "Referral bonus",
    "15,00",
    "KONTOSTAND AM 30.09.2020: 30,00 EUR",
    "Empowered by solarisBank AG",
    "HINWEIS ZUR EINLAGENSICHERUNG",
    "Guthaben sind als Einlagen nach Maßgabe des Einlagensicherungsgesetzes entschädigungsfähig. Nähere",
    "Informationen kannst Du dem Informationsbogen für den Einleger der solarisBank entnehmen. Diesen findest",
    "Du im Dokumenten-Center in der App."
  ]
]

```

</p></details>

---
Issue submitted with https://import.tresor.one

### LINKED ISSUE #1405: [] Reinvestierung
## Document
<details><summary>Abrechnung 6.pdf</summary><p>

```json
[
  [
    "TRADE REPUBLIC BANK GMBH",
    "KASTANIENALLEE 32",
    "10435 BERLIN",
    "Max Mustermann",
    "Musterstraße 1",
    "55555 Musterstad",
    "SEITE",
    "1",
    "von",
    "1",
    "DATUM",
    "28.01.2021",
    "DEPOT",
    "0163449201",
    "Trade Republic Bank GmbH",
    "Kastanienallee 32",
    "10435 Berlin",
    "www.traderepublic.com",
    "xxxxxxx@xxxxxxxxxxxxx.xxx",
    "Sitz der Gesellschaft: Düsseldorf",
    "AG Düsseldorf HRB 85864",
    "USt-ID DE307510626",
    "Geschäftsführer",
    "Andreas Willius",
    "Karsten Müller",
    "ABRE",
    "/ 28.01.2021",
    "/ 27404794",
    "/ 8f73-6e6e",
    "REINVESTIERUNG",
    "ÜBERSICHT",
    "NR.",
    "BUCHUNG",
    "WERTPAPIER",
    "BETRAG",
    "1",
    "Reinvestierung",
    "Main Street Capital Corp.",
    "Registered Shares DL -,01",
    "US56035L1044",
    "15 Stk.",
    "2",
    "Reinvestierung",
    "Main Street Capital Corp.",
    "Registered Shares DL -,01",
    "US56035L1044",
    "Stk.",
    "ABRECHNUNG",
    "NR.",
    "POSITION",
    "BETRAG",
    "1",
    "Bruttoertrag",
    "3,08 USD",
    "Quellensteuer für US-Emittent",
    "-0,46 USD",
    "GESAMT",
    "2,62 USD",
    "2",
    "Barausgleich",
    "3,08 USD",
    "Quellensteuer gem. Ertrag",
    "-0,46 USD",
    "Zwischensumme",
    "2,62 USD",
    "Zwischensumme",
    "1,21871 EUR/USD",
    "2,15 EUR",
    "GESAMT",
    "2,15 EUR",
    "BUCHUNG",
    "VERRECHNUNGSKONTO",
    "VALUTA",
    "BETRAG",
    "XX00000000000000000000",
    "15.01.2021",
    "4,77 USD",
    "Main Street Capital Corp. Registered Shares DL -,01 in Wertpapierrechnung in Deutschland.",
    "Lagerland: Vereinigte Staaten",
    "Diese Abrechnung wird maschinell erstellt und daher nicht unterschrieben.",
    "Sofern keine Umsatzsteuer ausgewiesen ist, handelt es sich gem. § 4 Nr. 8 UStG um eine umsatzsteuerfreie",
    "Leistung."
  ]
]

```

</p></details>

## Community profile

https://community.tresor.one/u/34993723

---
Issue submitted with https://import.tresor.one

### PR DESCRIPTION
**Pull request by SirGibihm**
* Throws correct status code for Account Statements Fixes #1134 
* Fixes #1405 where a new type of dividend has been proposed
* Added Testcase from #1405 
* Fixes #1105 (verifed via localhost:5000)

### CODE REVIEW

File: src/brokers/traderepublic.js [hunk 1]
```diff
@@ -156,53 +160,43 @@ const findTax = (textArr, fxRate) => {
...
+  } else if (
+    content.some(
+      line => line.includes('Sparplanausführung am') || line.includes('Kauf am')
+    )
+  ) {
+    return 'Buy';
+  } else if (content.some(line => line.includes('Verkauf am'))) {
+    return 'Sell';
+  }
+  console.log(content);
```
Comment (dredav) [inline src/brokers/traderepublic.js:198]: Did we need this console.log longer?

File: tests/brokers/__mocks__/traderepublic/index.js [hunk 2]
```diff
@@ -39,6 +40,7 @@ export const ignoredSamples = [
   require('./ignored/split.json'),
   require('./ignored/saving_plan_confirmation.json'),
   require('./ignored/saving_plan_change_confirmation.json'),
+  require('./ignored/account_statement.json'),
```
Comment (dredav) [inline tests/brokers/__mocks__/traderepublic/index.js:43]: Did we need a test for this document?

File: src/brokers/traderepublic.js (hunk 1 above)
Comment (SirGibihm) [reply src/brokers/traderepublic.js:198]: Absolutely not ^^' Removed

File: tests/brokers/__mocks__/traderepublic/index.js (hunk 2 above)
Comment (SirGibihm) [reply tests/brokers/__mocks__/traderepublic/index.js:43]: I thought all ignored statements would be automatically tested for TR. Fixed this

**Review by dredav — APPROVED** [review]

**Pull request merged**